openshift/adviser-template.yaml

```diff
@@ -0,0 +1,241 @@
+  - name: THOTH_ADVISER_LIBRARY_USAGE
+    required: false
+    description: Usage of libraries for adviser's resolution algorithm.
+    displayName: Library usage
+  - name: THOTH_ADVISER_REQUIREMENTS_FORMAT
+    required: false
+    description: Format of input
+    displayName: Requirements format
+  - name: THOTH_ADVISER_RECOMMENDATION_TYPE
+    required: false
+    description: Type of requested recommendation
+    displayName: Recommendation type
+  - name: THOTH_ADVISER_RUNTIME_ENVIRONMENT
+    required: false
+    description: Runtime environment to create recommendation for
+    displayName: Runtime environment
+  - name: THOTH_ADVISER_SEED
+    required: true
+    description: Random seed to run adviser with.
+    displayName: Random seed
+    # Keep seed constant across adviser runs so that two runs with same knowledge base report same results to user.
+    # Not to answer with a different stack each time a request is made.
+    value: "42"
+  - name: THOTH_ADVISER_PLOT
+    required: false
+    description: A path to history file to store annealing progress history.
+    displayName: History plot file
+    # TODO: Let's keep history for debugging purposes for now.
+    value: "history.png"
+  - name: THOTH_ADVISER_BEAM_WIDTH
+    required: true
+    description: Adviser's beam width to restrict memory requirements with states generated.
+    displayName: Beam width
+    value: "1000"
+  - name: THOTH_LOG_ADVISER
+    required: false
+    description: Log adviser actions
+    displayName: Log adviser actions
+    value: INFO
+  - name: THOTH_ADVISER_METADATA
+    required: false
+    description: Metadata carried with the adviser request
+    displayName: Adviser metadata
+    value: "null"
+  - name: THOTH_ADVISER_LIMIT
+    required: false
+    description: Limit number of software stacks scored.
+    displayName: Adviser limit.
+    value: "1000"
+  - name: THOTH_ADVISER_COUNT
+    required: false
+    description: Number of software stacks returned in the recommendation.
+    displayName: Adviser count.
+    value: "1"
+  - name: THOTH_ADVISER_LIMIT_LATEST_VERSIONS
+    required: false
+    description: Limit number of versions starting from latest in stacks.
+    displayName: Limit latest versions.
+    value: "5"
+  - name: THOTH_ADVISER_TIMEOUT_SECONDS
+    required: true
+    description: Give adviser specified number of seconds on CPU to compute results.
+    displayName: Adviser timeout seconds.
+    value: "3600"
+  - name: THOTH_FORCE_SYNC
+    description: Perform force sync - sync data even if they are reported to be existing
+    displayName: Perform force sync
+    required: true
+    value: "0"
+
+objects:
+  - apiVersion: argoproj.io/v1alpha1
+    kind: Workflow
+    metadata:
+      name: "${THOTH_ADVISER_JOB_ID}"
+      annotations:
+        thoth-station.ninja/template-version: 0.5.0
+      labels:
+        app: thoth
+        component: adviser
+    spec:
+      serviceAccountName: argo
```

Comment:
Why is this service account needed here?

Reply:
no I think no, we run all in `thoth-backend-stage` where all the workflows templates are placed

Reply:
Argo will run it with a default service account otherwise IIRC, which might not have sufficient privileges, ie. to list pods or get configmaps.

Reply:
Why adviser needs to list pods or configmaps?

Reply:
The underlying argoexec container might need to access the workflow controller configmap for example. Best is to try it w/o it and see how it behaves :)

Reply:
OK, so let's try to do it without argo sa. It sounds like an escalation of privileges.